pkg/query-service/app/http_handler.go

```diff
@@ -1854,6 +1856,38 @@ func (aH *APIHandler) deleteUser(w http.ResponseWriter, r *http.Request) {
 	aH.WriteJSON(w, r, map[string]string{"data": "user deleted successfully"})
 }
 
+// addUserFlag patches a user flags with the changes
+func (aH *APIHandler) patchUserFlag(w http.ResponseWriter, r *http.Request) {
+	// read user id from path var
+	userId := mux.Vars(r)["id"]
```

Comment:
Should this be taken from the token data instead of the URL var? Can I update someone else's if I know their ID from my account?

Reply:
You are right. I will change this.

Reply:
The permission on the path is SelfAccess. So this is not a problem.  

Though a better option would be what you suggest, i think we need to improve the self-access function to add user into the context. That will be useful caching of user instead of re-fetching it. 

```
func SelfAccess(f func(http.ResponseWriter, *http.Request)) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		user, err := auth.GetUserFromRequest(r)
```